examples/common/chip-app-server/server.cpp

```diff
@@ -0,0 +1,96 @@
+#include "server.h"
+
+#include <string.h>
+
+#include <inet/IPAddress.h>
+#include <inet/InetError.h>
+#include <inet/InetLayer.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <support/CodeUtils.h>
+#include <support/ErrorStr.h>
+#include <support/logging/CHIPLogging.h>
+#include <sys/param.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/UDP.h>
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::Transport;
+using namespace ::chip::DeviceLayer;
+
+#ifndef EXAMPLE_SERVER_NODEID
+// "nRF5"
+#define EXAMPLE_SERVER_NODEID 12344321
+#endif // EXAMPLE_SERVER_NODEID
+
+// Transport Callbacks
+namespace {
+extern "C" void HandleDataModelMessage(const MessageHeader & header, System::PacketBuffer * buffer, SecureSessionMgrBase * mgr);
+
+class ServerCallback : public SecureSessionMgrCallback
+{
+public:
+    virtual void OnMessageReceived(const MessageHeader & header, Transport::PeerConnectionState * state,
```

Comment:
I sort of wonder whether use of `git mv` would have led to a more informative diff here...  Ah, well.

Reply:
git mv is equivalent to delete + add, there is no explicit tracking. Renames are only detected at diff time by a heuristic based on % of lines changed (by default, if <50% of lines change in a delete/add, it's considered a rename).